container/container.go

```diff
@@ -58,136 +83,169 @@ func New(opts ...Option) *Container {
-func (c *Container) NonceUUID() (uuid.UUID, error) {
+//
+// See also SetNonceUUID.
+func (c Container) NonceUUID() (uuid.UUID, error) {
 	return uuid.FromBytes(c.v2.GetNonce())
 }
 
 // SetNonceUUID sets container nonce as UUID.
+//
+// See also NonceUUID.
 func (c *Container) SetNonceUUID(v uuid.UUID) {
 	data, _ := v.MarshalBinary()
 	c.v2.SetNonce(data)
 }
 
-func (c *Container) BasicACL() uint32 {
-	return c.v2.GetBasicACL()
+// BasicACL returns container's ACL.
+//
+// Zero Container has PrivateBasicRule ALC.
+//
+// See also SetBasicACL.
+func (c Container) BasicACL() acl.BasicACL {
+	return acl.BasicACL(c.v2.GetBasicACL())
 }
 
+// SetBasicACL sets basic ALC rule.
+//
+// See also BasicACL.
 func (c *Container) SetBasicACL(v acl.BasicACL) {
 	c.v2.SetBasicACL(uint32(v))
 }
 
-func (c *Container) Attributes() Attributes {
-	return NewAttributesFromV2(c.v2.GetAttributes())
+// Attributes return container's attributes.
+//
+// Zero Container has nil attributes.
+//
+// See also SetAttributes.
+func (c Container) Attributes() Attributes {
+	attrsV2 := c.v2.GetAttributes()
+	if attrsV2 == nil {
+		return nil
+	}
+
+	var attrs Attributes
+	attrs.ReadFromV2(attrsV2)
+
+	return attrs
 }
 
+// SetAttributes sets container's attributes.
+// Attributes must not be nil.
+//
+// See also Attributes.
 func (c *Container) SetAttributes(v Attributes) {
-	c.v2.SetAttributes(v.ToV2())
+	var attrsV2 []container.Attribute
+	v.WriteToV2(&attrsV2)
+
+	c.v2.SetAttributes(attrsV2)
 }
 
-func (c *Container) PlacementPolicy() *netmap.PlacementPolicy {
+// PlacementPolicy returns container's placement policy.
+//
+// Zero Container has nil policy.
+//
+// See also SetPlacementPolicy.
+func (c Container) PlacementPolicy() *netmap.PlacementPolicy {
 	return netmap.NewPlacementPolicyFromV2(c.v2.GetPlacementPolicy())
 }
 
+// SetPlacementPolicy sets placement policy.
+// Policy must not be nil.
+//
+// See also PlacementPolicy.
 func (c *Container) SetPlacementPolicy(v *netmap.PlacementPolicy) {
 	c.v2.SetPlacementPolicy(v.ToV2())
 }
 
 // SessionToken returns token of the session within
 // which container was created.
+//
+// Zero Container has nil token.
+//
+// See also SetSessionToken.
 func (c Container) SessionToken() *session.Token {
 	return c.token
 }
 
 // SetSessionToken sets token of the session within
 // which container was created.
+//
+// See also SessionToken.
 func (c *Container) SetSessionToken(t *session.Token) {
 	c.token = t
 }
 
 // Signature returns signature of the marshaled container.
+//
+// Zero Container has nil signature.
+//
+// See also SetSignature.
 func (c Container) Signature() *signature.Signature {
 	return c.sig
 }
 
 // SetSignature sets signature of the marshaled container.
+//
+// See also Signature.
 func (c *Container) SetSignature(sig *signature.Signature) {
 	c.sig = sig
 }
-
-// Marshal marshals Container into a protobuf binary form.
-func (c *Container) Marshal() ([]byte, error) {
```

Comment:
Really unused?

Reply:
if fact we discussed a little with @alexvanin that we gonna drop all marshaling methods and return them when doing SDK adapting since it is hard to track what is (un)used. but returned for container

as for my opinion, I'm still against saying that something *in a repo called SDK* is used and something is unused. for now, we are only able to say "it is unused in the node repo and the gateways", but this is not a library for them, somehow or other, we named it `SDK`

Reply:
yea, i understand your point, but can not agree with "unused code" term in `SDK` repo

> support of backward compatibility

can not imagine problems with marshaling structs (I mean anyway we have (and will always have) such methods for some structs. so we have to support them and I do not see problems with supporting marshaling for all of our structs. if the question is about the amount of code that must be changed if something is changed in the marshaling, I think that the current refactor is more labor-intensive but anyway we have agreed with that)